nav2_mppi_controller/include/nav2_mppi_controller/motion_models.hpp

```diff
@@ -71,42 +63,33 @@ class MotionModel
    */
   virtual void predict(models::State & state)
   {
-    // Previously completed via tensor views, but found to be 10x slower
-    // using namespace xt::placeholders;  // NOLINT
-    // xt::noalias(xt::view(state.vx, xt::all(), xt::range(1, _))) =
-    //   xt::noalias(xt::view(state.cvx, xt::all(), xt::range(0, -1)));
-    // xt::noalias(xt::view(state.wz, xt::all(), xt::range(1, _))) =
-    //   xt::noalias(xt::view(state.cwz, xt::all(), xt::range(0, -1)));
-    // if (isHolonomic()) {
-    //   xt::noalias(xt::view(state.vy, xt::all(), xt::range(1, _))) =
-    //     xt::noalias(xt::view(state.cvy, xt::all(), xt::range(0, -1)));
-    // }
-
     const bool is_holo = isHolonomic();
-    float max_delta_vx = model_dt_ * control_constraints_.ax_max;
-    float min_delta_vx = model_dt_ * control_constraints_.ax_min;
-    float max_delta_vy = model_dt_ * control_constraints_.ay_max;
-    float max_delta_wz = model_dt_ * control_constraints_.az_max;
-    for (unsigned int i = 0; i != state.vx.shape(0); i++) {
-      float vx_last = state.vx(i, 0);
-      float vy_last = state.vy(i, 0);
-      float wz_last = state.wz(i, 0);
-      for (unsigned int j = 1; j != state.vx.shape(1); j++) {
-        float & cvx_curr = state.cvx(i, j - 1);
+    float && max_delta_vx = model_dt_ * control_constraints_.ax_max;
+    float && min_delta_vx = model_dt_ * control_constraints_.ax_min;
+    float && max_delta_vy = model_dt_ * control_constraints_.ay_max;
+    float && max_delta_wz = model_dt_ * control_constraints_.az_max;
+
+    unsigned int n_rows = state.vx.rows();
+    unsigned int n_cols = state.vx.cols();
+
+    // Default layout in eigen is column-major, hence accessing elements in column-major fashion to utilize L1 cache as much as possible
+    for (unsigned int i = 1; i != n_cols; i++) {
+      for (unsigned int j = 0; j != n_rows; j++) {
+        float & vx_last = state.vx(j, i - 1);
+        float & cvx_curr = state.cvx(j, i - 1);
         cvx_curr = std::clamp(cvx_curr, vx_last + min_delta_vx, vx_last + max_delta_vx);
-        state.vx(i, j) = cvx_curr;
-        vx_last = cvx_curr;
```

Comment:
Readd the storing of last values, this is a good optimization since it removes lookups in the array (which are constant time, but not free-time)

Reply:
I'm accessing Array in column-major fashion, so the last value is of the different sampled batch.
By default, arrays are stored in memory in column-major order and Eigen is [optimized for column-major storage](https://eigen.tuxfamily.org/dox/group__TopicStorageOrders.html#:~:text=The%20default%20in%20Eigen%20is,best%20with%20column%2Dmajor%20matrices.) hence accessing elements contiguously reduces the cache misses of the CPU. 
I've tested them on a 4000*500 array, and accessing in column-major order was at least 2.5-3 times faster.

Reply:
````
#include <iostream>
#include <math.h>
#include <chrono>
#include <Eigen/Dense>

int main()
{
  Eigen::ArrayXXf vx1(4000, 500);
  vx1.setRandom();
  Eigen::ArrayXXf cvx1(4000, 500);
  cvx1.setRandom();

  Eigen::ArrayXXf vx2(4000, 500);
  vx2.setRandom();
  Eigen::ArrayXXf cvx2(4000, 500);
  cvx2.setRandom();
  float min_delta_vx = -0.2f;
  float max_delta_vx = 0.2f;

  std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
  
  for (unsigned int i = 1; i != 500; i++) {
    for (unsigned int j = 0; j != 4000; j++) {
      float & vx_last = vx1(j, i - 1);
      float & cvx_curr = cvx1(j, i - 1);
      cvx_curr = std::clamp(cvx_curr, vx_last + min_delta_vx, vx_last + max_delta_vx);
      vx1(j, i) = cvx_curr;
    }
  }

  std::chrono::steady_clock::time_point now = std::chrono::steady_clock::now();
  std::cout << "Time taken for column-major operation = " << std::chrono::duration_cast<std::chrono::microseconds>(now- begin).count() << "[us]" << std::endl;

  begin = std::chrono::steady_clock::now();

  for (unsigned int i = 0; i != 4000; i++) {
    float vx_last = vx2(i, 0);
    for (unsigned int j = 1; j != 500; j++) {
      float & cvx_curr = cvx2(i, j - 1);
      cvx_curr = std::clamp(cvx_curr, vx_last + min_delta_vx, vx_last + max_delta_vx);
      vx2(i, j) = cvx_curr;
      vx_last = cvx_curr;
    }
  }

  now = std::chrono::steady_clock::now();
  std::cout << "Time taken for Row-major operation = " << std::chrono::duration_cast<std::chrono::microseconds>(now- begin).count() << "[us]" << std::endl;
}
````